### PR TITLE
DIP1003: Remove body as a Keyword

### PR DESCRIPTION
**Pull request by WalterBright**
https://github.com/dlang/DIPs/blob/master/DIPs/DIP1003.md

It doesn't give a warning or deprecation yet over use of `body`.

### CODE REVIEW

**Review by UplinkCoder — APPROVED** [review]
nice one

File: src/ddmd/parse.d [hunk 1]
```diff
@@ -6618,13 +6629,19 @@ final class Parser(AST) : Lexer
                 }
                 return false;
 
+            case TOKidentifier:
```
Comment (andralex) [inline src/ddmd/parse.d]: these lines appear not covered

**Review by andralex — APPROVED** [review]
Please take a look at coverage. Thx!

File: src/ddmd/parse.d (hunk 1 above)
Comment (WalterBright) [inline src/ddmd/parse.d]: done

**Comment by WalterBright** [comment]
Auto-merge toggled on

**Comment by MetaLang** [comment]
Why was there no corresponding changelist entry for this? I just looked and there is no reference in the changelog to allowing `do`.

**Comment by ntrel** [comment]
@MetaLang I've added a related changelog entry in #7384, so at least the DIP will be mentioned in the next one.

File: test/fail_compilation/failcontracts.d [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+fail_compilation/failcontracts.d(22): Error: unexpected ( in declarator
+fail_compilation/failcontracts.d(22): Error: found 'T' when expecting ')'
+fail_compilation/failcontracts.d(22): Error: enum declaration is invalid
+fail_compilation/failcontracts.d(22): Error: found ')' instead of statement
+---
+*/
+
+void test()
+{
+    auto f1 = function() bode;
```
Comment (ibuclaw) [inline test/fail_compilation/failcontracts.d]: @WalterBright is this deliberate?  Or did you actually misspell `body`?

**Pull request merged**